src/Microsoft.AspNetCore.Authentication/AuthenticationMiddleware.cs

```diff
@@ -54,11 +54,16 @@ public AuthenticationMiddleware(RequestDelegate next, IAuthenticationSchemeProvi
                 var result = await context.AuthenticateAsync(defaultAuthenticate.Name);
                 if (result?.Principal != null)
                 {
+                    var identities = context.User?.Identities;
```

Comment:
HttpContext.User never returns null, it at least returns an anonymous identity.

Reply:
Good to know!